ext/pdo/pdo_sql_parser.re

```diff
@@ -270,40 +263,41 @@ safe:
 
 					switch (param_type) {
 						case PDO_PARAM_BOOL:
-							plc->quoted = zend_is_true(parameter) ? "1" : "0";
-							plc->qlen = sizeof("1")-1;
+							plc->quoted = zend_is_true(parameter) ? ZSTR_CHAR('1') : ZSTR_CHAR('0');
 							plc->freeq = 0;
 							break;
 
 						case PDO_PARAM_INT:
-							buf = zend_long_to_str(zval_get_long(parameter));
-
-							plc->qlen = ZSTR_LEN(buf);
-							plc->quoted = estrdup(ZSTR_VAL(buf));
+							plc->quoted = zend_long_to_str(zval_get_long(parameter));
 							plc->freeq = 1;
 							break;
 
 						case PDO_PARAM_NULL:
-							plc->quoted = "NULL";
-							plc->qlen = sizeof("NULL")-1;
+							plc->quoted = ZSTR_KNOWN(ZEND_STR_NULL);
 							plc->freeq = 0;
 							break;
 
-						default:
+						default: {
+							zend_string *quoted_buf;
+
+							/* TODO Should this be zval_try_get_string_func() ? */
```

Comment:
Just zval_try_get_string, but yes, that would be more elegant than the EG(exception) check.